File: smart_farm_app/lib/data/models/Livestock.dart
import '../models/Farm.dart';

class Livestock {
  int id;
  String livestockType;
  String livestockName;
  String imageUrl;
  String description;
  String breed;
  DateTime dateOfBirth;
  int farmId;
  Farm farm;
  String care;

  Livestock({
    required this.id,
    required this.livestockType,
    required this.livestockName,
    required this.imageUrl,
    required this.description,
    required this.breed,
    required this.dateOfBirth,
    required this.farmId,
    required this.farm,
    required this.care,
  });

  factory Livestock.fromJson(Map<String, dynamic> json) {
    return Livestock(
      id: json['id'],
      livestockType: json['livestockType'],
      livestockName: json['livestockName'],
      imageUrl: json['imageUrl'],
      description: json['description'],
      breed: json['breed'],
      dateOfBirth: DateTime.parse(json['dateOfBirth']),
      farmId: json['farmId'],
      farm: Farm.fromJson(json['farm']),
      care: json['livestockCares'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'livestockType': livestockType,
      'livestockName': livestockName,
      'imageUrl': imageUrl,
      'description': description,
      'breed': breed,
      'dateOfBirth': dateOfBirth.toIso8601String(),
      'farmId': farmId,
      'farm': farm.toJson(),
      'livestockCares': care,
    };
  }
}

File: SmartFarm/SmartFarm.Core/Entities/Livestock.cs
using SmartFarm.Core.Contracts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SmartFarm.Core.Entities
{
    public class Livestock : IEntity
    {
        public int Id { get; set; }

        public string LivestockType { get; set; }

        public string LivestockName { get; set; }

        public string ImageUrl { get; set; }

        public string Description { get; set; }

        public string Breed { get; set; }

        public string Care { get; set; }

        public DateTime DateOfBirth { get; set; }

        public int FarmId { get; set; }

        public Farm Farm { get; set; }

        public List<Report> Reports { get; set; }
    }
}
